subsys/net/lib/lwm2m/lwm2m_message_handling.c

```diff
@@ -446,6 +446,10 @@ void lwm2m_engine_context_close(struct lwm2m_ctx *client_ctx)
 
 	for (i = 0, msg = messages; i < ARRAY_SIZE(messages); i++, msg++) {
 		if (msg->ctx == client_ctx) {
+			/* Timeout "send" method messages pending*/
+			if (msg->message_timeout_cb) {
```

Comment:
Wouldn't [send_status_cb()](https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/net/lib/lwm2m/lwm2m_object.h#L515C18-L515C32) with `LWM2M_SEND_STATUS_FAILURE` make more sense here? Timeout is not really what happens here.

Reply:
You are right.
I suggest:
- Update static void do_send_timeout_cb(struct lwm2m_message *msg) to accept an enum as input argument

or 

- Update this function: static int do_send_reply_cb(const struct coap_packet *response, struct coap_reply *reply,
			    const struct sockaddr *from) to accept a specific code

Reply:
But why not simply call `msg->send_status_cb(LWM2M_SEND_STATUS_FAILURE);` directly here?

Reply:
Sometimes I make life harder :)